### DISCUSSION TITLE
Why does `kamal build` take a deploy lock?

### DISCUSSION THREAD

**Discussion by clintmiller** [basecamp/kamal#446] [Q&A]
Is there a specific reason that calling `kamal build` takes the deploy lock? It seems like it should be generally safe for builds to run concurrently, regardless of the environment in which they are executed.

Surely I'm just overlooking something simple?

**Comment by djmb** [comment]
Hi @clintmiller - thanks for pointing this out! I'm not sure there is a good reason. I'll convert this to an issue to track it.
